src/async_reader/zarr_read_async.rs

```diff
@@ -32,16 +31,16 @@ pub struct ZarrPath {
     location: Path,
 }
 
-impl ZarrPath{
+impl ZarrPath {
     pub fn new(store: Arc<dyn ObjectStore>, location: Path) -> Self {
-        Self {store, location}
+        Self { store, location }
     }
 }
 
 /// Implementation of the [`ZarrReadAsync`] trait for a [`ZarrPath`] which contains the
 /// object store that the data will be read from.
 #[async_trait]
-impl ZarrReadAsync for ZarrPath {
+impl<'a> ZarrReadAsync<'a> for ZarrPath {
```

Comment:
So, is adding the lifetimes and replacing vectors with slices something clippy complains about? Oh, wait adding the lifetime is _because_ the vectors were replaced with slices right? Do you know why it doesn't like vectors there, I mean sure they might have a tiny bit of overhead, but still...?

Reply:
This is the error clippy shows: https://rust-lang.github.io/rust-clippy/master/index.html#/ptr_arg

As an aside, if you don't have clippy setup, it's pretty easy to use/run and obviously after this PR is merged it'll be required.

Reply:
Yep, I'll set it up before my next merge.